### PR TITLE
Generate daily report for new email brands

### PR DESCRIPTION
**Pull request by samuelhwilliams**
Re-creation of https://github.com/alphagov/notifications-api/pull/3647

Now with an added test to detect/check that HTMLEmailTemplate can be handled for services with an email brand.

### CODE REVIEW

**Comment by samuelhwilliams** [comment]
This has sort-of already been reviewed by @klssmith (thank you) but there's a new test for the weird edge case with the new HTMLEmailTemplate now

File: app/celery/scheduled_tasks.py [hunk 1]
```diff
@@ -375,3 +380,68 @@ def delete_old_records_from_events_table():
...
+        domain=current_app.config["ADMIN_BASE_URL"],
+        yesterday=previous_weekday.strftime("%A %-d %B %Y"),
+        brands_by_organisation=brands_by_organisation,
+        brands_with_no_organisation=brands_with_no_organisation,
+    )
+
+    ticket = NotifySupportTicket(
+        subject="Review new email brandings",
+        message=message,
+        ticket_type=NotifySupportTicket.TYPE_INCIDENT,
```
Comment (quis) [inline app/celery/scheduled_tasks.py:443]: Is `TYPE_INCIDENT` appropriate here? Might [`TYPE_TASK`](https://github.com/alphagov/notifications-utils/blob/main/notifications_utils/clients/zendesk/zendesk_client.py#L50) be better?

File: app/celery/scheduled_tasks.py [hunk 2]
```diff
@@ -375,3 +380,68 @@ def delete_old_records_from_events_table():
...
+    brands_with_no_organisation = []
+    for new_brand in new_email_brands:
+        if not new_brand.organisations:
+            brands_with_no_organisation.append(new_brand)
+
+        else:
+            for organisation in new_brand.organisations:
+                brands_by_organisation[organisation].append(new_brand)
+
+    template = jinja2.Template(
```
Comment (quis) [inline app/celery/scheduled_tasks.py:412]: Could you pull this out into its own file please? Will make it easier to edit in the future when we add some guidance from https://docs.google.com/document/d/17tZKy0ogUOmUkjydjMMRTzEVZzpBrJFQEf3k789o3gs/edit#heading=h.xficsb7mfn5u

File: app/celery/scheduled_tasks.py (hunk 1 above)
Comment (samuelhwilliams) [reply app/celery/scheduled_tasks.py:443]: I had it as type_task originally but it was suggested we rarely use that and incident is more common for us.

File: tests/app/delivery/test_send_to_providers.py [hunk 3]
```diff
@@ -393,6 +393,23 @@ def test_send_email_should_use_service_reply_to_email(sample_service, sample_ema
     )
 
 
+@pytest.mark.parametrize("service_fixture", ["sample_service", "sample_service_with_email_branding"])
+def test_send_email_works_with_and_without_email_branding(request, service_fixture, sample_email_template, mocker):
+    service = request.getfixturevalue(service_fixture)
+    mocker.patch("app.aws_ses_client.send_email", return_value="reference")
+
+    db_notification = create_notification(template=sample_email_template, reply_to_text="foo@bar.com")
+    create_reply_to_email(service=service, email_address="foo@bar.com")
```
Comment (klssmith) [inline tests/app/delivery/test_send_to_providers.py:402]: The tests pass without this line, so I don't think we need to create the reply_to email address for the test

File: app/celery/scheduled_tasks.py (hunk 1 above)
Comment (samuelhwilliams) [reply app/celery/scheduled_tasks.py:424]: Have switched to TYPE_TASK on second thought, it really does just feel more appropriate...

File: tests/app/conftest.py [hunk 4]
```diff
@@ -157,6 +157,14 @@ def sample_service(sample_user):
     return service
 
 
+@pytest.fixture(scope="function")
+def sample_service_with_email_branding(sample_service):
+    sample_service.email_branding = create_email_branding(
+        id=uuid.uuid4(), logo="1ac6f483-3105-4c9e-9017-dd7fb2752c44-nhs-blue_x2.png", name="NHS"
```
Comment (klssmith) [inline tests/app/conftest.py:163]: You could optionally use a non-NHS example for this fixture just in case someone tries to use this along with the `nhs_email_branding` fixture in the same test and gets a surprising integrity error - but that's probably very unlikely to happen.

File: tests/app/delivery/test_send_to_providers.py (hunk 3 above)
Comment (samuelhwilliams) [reply tests/app/delivery/test_send_to_providers.py:402]: Patched in fixup commit 5da28b6b9ed665cef0dbe809bb83c286ea0c9b43

File: tests/app/conftest.py (hunk 4 above)
Comment (samuelhwilliams) [reply tests/app/conftest.py:163]: Patched in fixup commit 5da28b6b9ed665cef0dbe809bb83c286ea0c9b43

**Review by klssmith — APPROVED** [review]
